x-pack/plugins/security/server/spaces/secure_spaces_client_wrapper.ts

```diff
@@ -265,6 +269,24 @@ export class SecureSpacesClientWrapper implements ISpacesClient {
       }
     }
 
+    // Fetch saved objects to be removed for audit logging
+    const finder = this.spacesClient.createSavedObjectFinder(id);
+    for await (const response of finder.find()) {
+      response.saved_objects.forEach((savedObject) => {
+        const isOnlySpace = !savedObject.namespaces || savedObject.namespaces.length === 1;
+        this.auditLogger.log(
+          savedObjectEvent({
+            action: isOnlySpace
+              ? SavedObjectAction.DELETE
+              : SavedObjectAction.UPDATE_OBJECTS_SPACES,
+            outcome: 'unknown',
+            savedObject: { type: savedObject.type, id: savedObject.id },
+          })
+        );
```

Comment:
The `UPDATE_OBJECTS_SPACES` audit event should include the space that the object is being removed from in the `deleteFromSpaces` field, for example:

https://github.com/elastic/kibana/blob/8a39a113a0426a60df7b30731133dec1cc922c1f/x-pack/plugins/security/server/saved_objects/secure_saved_objects_client_wrapper.ts#L854-L862